### PR TITLE
Add accessibility linting

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5155 (for now; more could probably be done).

### Description (What does it do?)
This PR adds [eslint-plugin-styled-components-a11y](https://www.npmjs.com/package/eslint-plugin-styled-components-a11y?activeTab=readme) to lint accessibility issues. This package is a wrapper around [eslint-plugin-jsx-a11y](https://www.npmjs.com/package/eslint-plugin-jsx-a11y).

### How can this be tested?
- Linting should pass
- Try removing `aria-label` from an ActionButton; it should give you a useful error message
- Try removing an alt tag from an `img`, or any `styled.img`. You should get a useful error message.

### Additional Context
One of the main user-facing changes here is that I set many alt tags to `""` (e.g., on testimonials). This is better than no alt tag, and IMO is appropriate. These images are generally decorative, not, e.g., diagrams or a literal, physical product. And the testimonial APIs don't provide alt data, so we don't really have another choice.

### CODE REVIEW

File: frontends/ol-components/src/components/ShareTooltip/ShareTooltip.tsx
Comment (ChristopherChudzicki) [inline frontends/ol-components/src/components/ShareTooltip/ShareTooltip.tsx:1]: There was a linting error in ShareTooltip.

We don't use this component. We have a "Share" button in our drawer designs (not implemented). I strongly suspect we would implement it a bit differently than it was here anyway. E.g., we would use MUI popover not `rc-tooltip`.

**Comment by jonkafton** [comment]
> One of the main user-facing changes here is that I set many alt tags to `""` (e.g., on testimonials). This is better than no alt tag, and IMO is appropriate. These images are generally decorative

Agree and not sure if it would be feasible, or even useful, to describe image content, e.g. "a woman on a laptop concentrating"

**Comment by jonkafton** [comment]
Looks good, works well. Interestingly, it didn't pick on removing the `aria-label` on `src/page-components/SearchDisplay/SearchDisplay.tsx:667` - a styled ol-component `Button`. We pass props all the way through,  but maybe a restriction on two levels of styled() wrapping?

**Comment by ChristopherChudzicki** [comment]
>  Interestingly, it didn't pick on removing the aria-label on src/page-components/SearchDisplay/SearchDisplay.tsx:667 - a styled ol-component Button. We pass props all the way through, but maybe a restriction on two levels of styled() wrapping?

I think more likely a bug with [eslint-plugin-styled-components-a11y](https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y) wrapper for [eslint-plugin-jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y)

**Review by jonkafton — APPROVED** [review]

**Pull request merged**